### PR TITLE
Unique nesting of Platform/Instrument using group by

### PR DESCRIPTION
**Pull request by sachitrajbhandari**
https://github.com/aodn/backlog/issues/1627

### CODE REVIEW

**Comment by sachitrajbhandari** [comment]
> This corrects an issue with the mcp data model where platform /instrument associations were denormalised.

Fixed as suggested.

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+            <mrc:MD_AttributeGroup>
+               <mrc:contentType>
+                  <mrc:MD_CoverageContentTypeCode codeList="http://standards.iso.org/iso/19115/resources/Codelists/cat/codelists.xml#MD_CoverageContentTypeCode"
+                                                  codeListValue="physicalMeasurement"/>
+               </mrc:contentType>
+               <mrc:attribute>
+                  <mrc:MD_SampleDimension>
+                     <mrc:name>
+                        <mcc:MD_Identifier>
+                           <mcc:authority/>
```
Comment (jonescc) [inline geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml:274]: Not required - this element is optional  and no vocabularyPubisher is defined (we don't use it and I don't think anyone else does either)

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+                           <mcc:code>
+                              <gcx:Anchor xlink:href="http://vocab.nerc.ac.uk/collection/P01/current/TEMPPR01">Temperature of the water body</gcx:Anchor>
+                           </mcc:code>
+                           <mcc:version gco:nilReason="missing">
```
Comment (jonescc) [inline geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml:278]: Not required - this element is optional and no vocabularyversion is defined (we don't use it and I don't think anyone  else does either).

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+                              <gco:CharacterString/>
+                           </mcc:version>
+                        </mcc:MD_Identifier>
+                     </mrc:name>
+                     <mrc:units>
+                        <gml:BaseUnit gml:id="d19e13">
+                           <gml:descriptionReference/>
```
Comment (jonescc) [inline geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml:285]: This element is optional so don't need an empty element.  Could map termUri for the unit to it as its a reference to a remote description of the unit

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml (hunk 1 above)
Comment (sachitrajbhandari) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml]: Removed <mcc:authority/>

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml (hunk 3 above)
Comment (sachitrajbhandari) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml]: Removed <gml:descriptionReference/>

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml (hunk 2 above)
Comment (sachitrajbhandari) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml]: Removed <mcc:version gco:nilReason="missing">

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml (hunk 1 above)
Comment (sachitrajbhandari) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml]: Removed <mcc:authority/>

**Comment by sachitrajbhandari** [comment]
> Looks good but we can we ensure the input/output metadata.xml validates.

XML records validated.

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+   <mdb:identificationInfo>
+      <mri:MD_DataIdentification>
+         <mri:descriptiveKeywords>
+            <mri:MD_Keywords>
+               <mri:keyword>
+                  <gcx:Anchor xlink:href="http://vocab.nerc.ac.uk/collection/L06/current/25">unmanned autonomous underwater vehicle</gcx:Anchor>
+               </mri:keyword>
+               <mri:type>
+                  <mri:MD_KeywordTypeCode codeList="http://standards.iso.org/iso/19115/resources/Codelists/cat/codelists.xml#MD_KeywordTypeCode"
+                                          codeListValue="theme"/>
```
[the remark this answers is not in the document]
Comment (jonescc) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml:60]: Missed this one?

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+               </mri:keyword>
+               <mri:keyword>
+                  <gcx:Anchor xlink:href="http://vocab.aodn.org.au/def/instrument/entity/361">water biogeochemical sensor</gcx:Anchor>
+               </mri:keyword>
+               <mri:keyword>
+                  <gcx:Anchor xlink:href="http://vocab.nerc.ac.uk/collection/L05/current/379">altimeter</gcx:Anchor>
+               </mri:keyword>
+               <mri:type>
+                  <mri:MD_KeywordTypeCode codeList="http://standards.iso.org/iso/19115/resources/Codelists/cat/codelists.xml#MD_KeywordTypeCode"
+                                          codeListValue="theme"/>
```
[the remark this answers is not in the document]
Comment (jonescc) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml:198]: And this one?

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (sachitrajbhandari) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml:60]: The value of codeListValue attribute depends on the vocabularies used.

File: geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml (hunk 5 above)
[the remark this answers is not in the document]
Comment (sachitrajbhandari) [reply geonetwork/transform-mcp-iso19115-3/src/test/resources/acquisitionInformation/expected/af5d0ff9-bb9c-4b7c-a63c-854a630b6984.xml:198]: The value of codeListValue attribute depends on the vocabularies used.

**Pull request merged**